client/Containers/FilterSortControl.jsx

```diff
@@ -0,0 +1,188 @@
+//     }
+//     if (clear) {
+//       dispatch(clearFilters());
+//     }
+//     dispatch(resetPagination());
+//   },
+// });
+
+const FilterSortBar = styled.div`
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: center;
+  align-items: center;
+  background: black;
+  width: calc(100% - 0.5rem);
+  border-radius: 0 0 0.5rem 0.5rem;
+  color: white;
+  padding: 0 0.5rem 0.5rem 0;
+`;
+
+const AllFilterContainer = styled.div`
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: center;
+  align-items: center;
+`;
+
+const FilterContainer = styled.div`
+  display: flex;
+`;
+
+const FilterLabel = styled.label`
+  padding: 0.5rem 0 0.5rem 0.5rem;
+`;
+
+const FilterSelect = styled.select`
+  margin: 0.5rem;
+`;
+
+const FilterSortControl = () => {
+  const { productFilters, productPagination } = useSelector(
+    ({ productFilters, productPagination }) => ({
+      productFilters,
+      productPagination,
+    })
+  );
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const getProducts = (query) => dispatch(fetchProductList(query));
+  const filter = (options) => {
+    // options: { type?: string, style?: string, room?: string, clear?: true|false }
+    for (const key in productFilters) {
+      if (typeof options[key] === 'string') productFilters[key] = options[key];
+    }
+    if (options.clear) {
+      for (const key in productFilters) {
+        productFilters[key] = '';
+      }
+    }
+    const query = buildProductQuery({
+      productFilters,
+      productPagination: { ...productPagination, page: 1 },
+    });
+    history.push(`/products${query}`);
+    getProducts(query);
+  };
+  const paginate = (options) => {
+    // options: { page?: INT, size?: INT, sort?: STRING }
+    for (const key in productPagination) {
+      if (options[key]) productPagination[key] = options[key];
+    }
+    const query = buildProductQuery({ productFilters, productPagination });
+    history.push(`/products${query}`);
+    getProducts(query);
+  };
+  return (
+    <FilterSortBar>
+      <FilterContainer>
+        <FilterLabel>SORT</FilterLabel>
+        <FilterSelect
+          value={productPagination.sort}
+          onChange={(ev) => paginate({ page: 1, sort: ev.target.value })}
+        >
+          <option value={'name,ASC'}>Name, A-Z</option>
+          <option value={'name,DESC'}>Name, Z-A</option>
+          <option value={'price,DESC'}>Price, High to Low</option>
+          <option value={'price,ASC'}>Price, Low to High</option>
+        </FilterSelect>
+      </FilterContainer>
+      <FilterContainer>
+        <FilterLabel>SIZE</FilterLabel>
+        <FilterSelect
+          value={productPagination.size}
+          onChange={(ev) => paginate({ page: 1, size: ev.target.value * 1 })}
+        >
+          <option value="6">6 per Page</option>
+          <option value="12">12 per Page</option>
+          <option value="18">18 per Page</option>
+          <option value="24">24 per Page</option>
```

Comment:
Also just some initial page size options, open to expanding, modifying or reducing these.